site/app/models/CalendarInfo.php

```diff
@@ -59,13 +61,22 @@ class CalendarInfo extends AbstractModel {
      * @param array $gradeables_of_user container of gradeables in the system
      * @return CalendarInfo
      */
-    public static function loadGradeableCalendarInfo(Core $core, array $gradeables_of_user): CalendarInfo {
+    public static function loadGradeableCalendarInfo(Core $core, array $gradeables_of_user, array $courses, array $calendar_items): CalendarInfo {
         $info = new CalendarInfo($core);
         $date_format = $core->getConfig()->getDateTimeFormat()->getFormat('gradeable');
 
         $gradeable_list = new GradeableList($core, $core->getUser(), $gradeables_of_user["gradeables"]);
 
-        // get the gradeables from the GradeableList and group them by section
+        $i = 1;
+        /** @var Course $course */
+        foreach ($courses as $course) {
+            $info->colors[$course->getSemester() . $course->getTitle()] = "var(--category-color-$i)";
+            if ($i < 21) {
```

Comment:
Where does the magic number 21 come from?  Constants like this are generally a bad thing for code readability.

Reply:
Its for the colors that calendar currently implements:
<img width="211" alt="image" src="https://user-images.githubusercontent.com/114759691/229907174-16a3c4fe-9f88-4e7b-b1a0-c6078b0c4440.png">